restservice/src/main/java/org/itevents/util/mail/MailBuilderUtil.java

```diff
@@ -33,69 +32,78 @@
-        return buildMailFromXmlUserOtp(BuildXmlFromUserOtp(user, oneTimePassword));
+    public String buildHtmlFromUserOtp(User user, OneTimePassword oneTimePassword) {
+        try{
+            return buildMailFromXmlUserOtp(BuildXmlFromUserOtp(user, oneTimePassword));
+        } catch (JAXBException | TransformerException | IOException e) {
+        throw new MailBuilderUtilException(e.getMessage(), e);
+        }
     }
 
     private String buildXmlFromEventList(List<Event> events) throws JAXBException {
         EventsXmlNodeWrapper eventsXmlNodeWrapper = new EventsXmlNodeWrapper();
         eventsXmlNodeWrapper.setEvents(events);
-
-        Marshaller marshaller = JAXBContext.newInstance(EventsXmlNodeWrapper.class)
-                .createMarshaller();
-        marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
-        marshaller.setProperty(Marshaller.JAXB_ENCODING, "UTF-8");
-
-        StringWriter eventsInXmlStringWriter = new StringWriter();
-        marshaller.marshal(eventsXmlNodeWrapper, eventsInXmlStringWriter);
-        return eventsInXmlStringWriter.toString();
+            Marshaller marshaller = JAXBContext.newInstance(EventsXmlNodeWrapper.class)
+                    .createMarshaller();
+            marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
+            marshaller.setProperty(Marshaller.JAXB_ENCODING, "UTF-8");
+
+            StringWriter eventsInXmlStringWriter = new StringWriter();
+            marshaller.marshal(eventsXmlNodeWrapper, eventsInXmlStringWriter);
+            return eventsInXmlStringWriter.toString();
+//        }
     }
 
     private String buildMailFromXmlEvents(String eventsInXml) throws IOException, TransformerException {
-        Transformer transformer =
-                TransformerFactory.newInstance().newTransformer(
-                        new StreamSource(emailTemplateXslResource.getFile())
-                );
-
-        StringWriter mailStringWriter = new StringWriter();
-        transformer.transform(
-                new StreamSource(new StringReader(eventsInXml)),
-                new StreamResult(mailStringWriter)
-        );
+            Transformer transformer =
+                    TransformerFactory.newInstance().newTransformer(
+                            new StreamSource(emailTemplateXslResource.getFile())
+                    );
+
+            StringWriter mailStringWriter = new StringWriter();
+            transformer.transform(
+                    new StreamSource(new StringReader(eventsInXml)),
+                    new StreamResult(mailStringWriter)
+            );
+            return mailStringWriter.toString();
 
-        return mailStringWriter.toString();
     }
 
-    private String BuildXmlFromUserOtp(User user, OneTimePassword oneTimePassword) throws JAXBException, IOException {
+    private String BuildXmlFromUserOtp(User user, OneTimePassword oneTimePassword) throws JAXBException {
```

Comment:
@romach i'm trying to make this like template code, as in other methods in this class